kafka-inmemory/Dockerfile

```diff
@@ -1,26 +1,10 @@
-FROM fedora:25
+FROM enmasseproject/kafka-base:latest
 
-RUN dnf -y install which java-1.8.0-openjdk libaio python gettext hostname iputils wget && dnf clean all -y
+EXPOSE 9092
 
-# set Scala and Kafka version
-ENV SCALA_VERSION=2.11
-ENV KAFKA_VERSION=0.11.0.0
-# set Kafka home folder
-ENV KAFKA_HOME=/opt/kafka
-
-# Set from build args
-ARG version=latest
-ENV VERSION ${version}
-
-# downloading/extracting Apache Kafka
-RUN wget http://www.eu.apache.org/dist/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz
-
-RUN mkdir $KAFKA_HOME \
-    && tar xvfz kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz -C $KAFKA_HOME --strip-components=1
-
-WORKDIR $KAFKA_HOME
-
-# copy template configuration files
-COPY ./config/zookeeper.properties.template $KAFKA_HOME/config/
 # copy scripts for starting Kafka and Zookeeper
 COPY ./scripts/ $KAFKA_HOME
+
+USER kafka:kafka
+
+ENTRYPOINT ["/opt/kafka/kafka_run.sh"]
```

Comment:
Genuine question: Why is it better to use a run script like this than put the command line in the `.yml` for the replicaset/statefulset?

Reply:
My two cents ... 
* Leaving Kubernetes / Openshift aside ... setting properly the `ENTRYPOINT` / `CMD` in docker image makes it a lot easier for users to run the image. They can just type `docker run ...` without searching for the rights script to execute.
* In the Openshift / Kubernetes yaml files we already use the Zookeeper image on two places. So instead of specifying the zookeeper_run.sh command once in the Docker image we would already have the `command` duplicated twice in the repo.
* I think it leads a bit to cleaner design of the Docker images. If you have a single purpose docker images than it should be easy for you to have a single entrypoint and the command is not needed.